webapp/src/components/time_selector.tsx

```diff
@@ -1,48 +1,83 @@
-import React, {useMemo} from 'react';
+import React, {useEffect, useMemo} from 'react';
 import {useSelector} from 'react-redux';
 
 import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
 
 import ReactSelectSetting from './react_select_setting';
 
+const minuteStep = 15;
+
 type Props = {
     value: string;
     onChange: (value: string) => void;
-
-    // TODO: implement upper bound and lower bound to make sure end > start
-    // upperBound: idk;
-    // lowerBound: idk;
+    startTime?: string
+    endTime?: string
 }
 
 export default function TimeSelector(props: Props) {
     const theme = useSelector(getTheme);
+    let options = null;
+    let value = null;
+    let ranges: string[];
 
-    const options = useMemo(() => militaryTimeOptions.map(t => ({
-        label: t,
-        value: t,
-    })), []);
+    const updateOptions = () => {
+        let fromHour, fromMinute = 0;
+        let toHour = 23;
+        let toMinute = 45;
 
-    let value = null;
-    if (props.value) {
-        value = options.find(option => option.value === props.value);
+        if (props.startTime != undefined && props.startTime != '') {
+            const parts = props.startTime.split(":")
+            fromHour = parseInt(parts[0]);
+            fromMinute = parseInt(parts[1]) + minuteStep;
+            ranges = generateMilitaryTimeArray(fromHour, fromMinute, toHour, toMinute)
+        }
```

Comment:
> When I read this I'm thinking "Adding 15 will overflow into the next hour", and it looks like that case is actually taken care of in the `generateMilitaryTimeArray` function. The coupling of logic between those is a bit concerning. Can we instead adjust `fromHour` and `fromMinute` here?

What case do you want to cover here? That having a `fromMinute = 0` get's converted to 45?

Reply:
It's probably fine the way it is :+1: I'm not sure what I'm after here, it's just a bit hard to read exactly what's going on since there's logic in multiple places